src/main/java/com/google/sps/data/SmallCityService.java

```diff
@@ -38,23 +38,28 @@ public SmallCityService() { }
   * Create UserService instance from zipCode and get businesses list
   * @param zipCode inputted zipcode of user
   **/
-  public void createUserServiceWithZip(String zipCode) {
+  public void createUserServiceWithZip(String zipCode, String product) {
     this.userService = new UserService();
-    userService.createUserWithZipCode(zipCode);
+    userService.createUserWithZipCode(zipCode, product);
```

Comment:
I believe you have a SearchObject class that would be much better suited for this - containing the product search string that was use. 
ref: https://github.com/googleinterns/SmallCity/blob/master/src/main/java/com/google/sps/data/SearchObject.java

Reply:
I implemented the SearchObject class and removed the instance from the user service. Now, the searchObject only exists in the DataServlet and smallCityService.